nofish/NF_ReaScript.cpp

```diff
@@ -479,3 +480,20 @@ bool NF_DeleteTakeFromItem(MediaItem* item, int takeIdx)
 	SNM_TakeParserPatcher takePatcher(item);
 	return takePatcher.RemoveTake(takeIdx);
 }
+
+// Base64
+void NF_Base64_Decode(const char* base64_str, char* decodedStrOut, int decodedStrOut_sz)
+{
+	Base64 b64; int bufsize;
+	char* decoded = b64.Decode(base64_str, &bufsize); // bufsize holds the decoded length
+	if (realloc_cmd_ptr(&decodedStrOut,&decodedStrOut_sz, bufsize+1)) // resize REAPER's given buffer (incl. null terminator)
```

Comment:
The buffer is no longer null-terminated after using `realloc_cmd_ptr` (it becomes part of the string).

https://github.com/cfillion/reaimgui/blob/master/api/input.cpp#L27-L39

Reply:
Thanks, PR updated.
Dealing with (C-)strings gives me a hard time somehow :/, so further guidance would be appreciated.

I tried to use/adapt your function (to take a const char* instead of std::string, as I thought that fits better for general ReaScript usage), but can't get it to work correctly.

![meVsCstrings](https://user-images.githubusercontent.com/7658194/179791072-cf7ee14a-2d00-4b5f-b450-8af7e9d636ce.jpg)